import { NextRequest, NextResponse } from 'next/server'
import { ApiError } from './apiError'

export async function handleApiError(error: unknown) {
	console.error('‚ùå Error:', error)

	if (error instanceof ApiError) {
		return NextResponse.json(
			{
				message: error.message,
			},
			{ status: error.status }
		)
	}

	return NextResponse.json(
		{
			message: 'Internal Server Error',
		},
		{ status: 500 }
	)
}
